ksql-engine/src/main/java/io/confluent/ksql/function/FunctionNameValidator.java

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright 2018 Confluent Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.confluent.ksql.function;
+
+import com.google.common.collect.ImmutableSet;
+
+import java.util.Set;
+import java.util.function.Predicate;
+
+/**
+ * Check that a function name is valid. It is valid if it is not a Java reserved word
+ * and not a ksql reserved word and is a valid java identifier.
+ */
+class FunctionNameValidator implements Predicate<String> {
+  private static final Set<String> JAVA_RESERVED_WORDS
+      = ImmutableSet.<String>builder()
+      .add("abstract").add("assert").add("boolean").add("break").add("byte").add("case")
+      .add("catch").add("char").add("class").add("const").add("continue").add("default")
+      .add("double").add("else").add("enum").add("extends").add("do").add("final").add("finally")
+      .add("float").add("for").add("goto").add("if").add("int").add("implements").add("import")
+      .add("instanceof").add("interface").add("long").add("native").add("new").add("package")
+      .add("private").add("public").add("protected").add("return").add("this").add("throw")
+      .add("throws").add("transient").add("try").add("short").add("static").add("strictfp")
+      .add("super").add("switch").add("synchronized").add("void").add("volatile").add("while")
+      .build();
+
+  // This has been generated from SqlBase.tokens
+  private static final Set<String> KSQL_RESERVED_WORDS
+      = ImmutableSet.<String>builder()
+      .add("add").add("advance").add("all").add("alter").add("analyze").add("and").add("any")
+      .add("approximate").add("array").add("as").add("asc").add("asterisk").add("at")
+      .add("backquoted_identifier").add("beginning").add("bernoulli").add("between")
```

Comment:
'bernoulli', 'cross,  'nfd', 'nkfc', etc. etc. this makes me think that the list was autogenerated from the lexer tokens. There is way to get this automatically from the `SqlBaseParser.vocabulary`public static field. That would be a more robust way to do this IMO.

Reply:
Cool - thanks. Should've looked there